### PR TITLE
Update to hooks

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: src/components/date.js [hunk 1]
```diff
@@ -4,44 +4,50 @@
...
+  const { formatDate, textComponent: Text } = useIntl()
+
+  const { value, children } = props
+
+  return useMemo(() => {
+    let formattedDate = formatDate(value, props)
 
     if (typeof children === 'function') {
-      return children(formattedDate);
+      return children(formattedDate)
```
Comment (taion) [inline src/components/date.js]: sorta wonder if these are still relevant... like, wouldn't a `useFormattedDate` hook give us equivalent functionality? i guess it doesn't hurt

File: src/components/date.js [hunk 2]
```diff
@@ -4,44 +4,50 @@
...
     }
 
-    return <Text>{formattedDate}</Text>;
-  }
+    return <Text>{formattedDate}</Text>
+  }, [
+    value,
+    children,
+    props.localeMatcher,
```
Comment (taion) [inline src/components/date.js:28]: meh. i guess we strip the prop types in prod, so we can't use those values?

File: src/components/message.js [hunk 3]
```diff
@@ -4,159 +4,144 @@
...
+    // Creates a token with a random UID that should not be guessable or
+    // conflict with other parts of the `message` string.
+    let uid = Math.floor(Math.random() * 0x10000000000).toString(16)
+
+    let generateToken = (() => {
+      let counter = 0
+      return () => `ELEMENT-${uid}-${(counter += 1)}`
+    })()
+
+    // Splitting with a delimiter to support IE8. When using a regex
```
Comment (taion) [inline src/components/message.js:78]: do we still support IE8?

File: src/components/message.js [hunk 4]
```diff
@@ -4,159 +4,144 @@
...
+    tagName: Component = Text,
+  } = props
+
+  let tokenDelimiter
+  let tokenizedValues
+  let elements
+
+  let hasValues = values && Object.keys(values).length > 0
+  if (hasValues) {
+    // Creates a token with a random UID that should not be guessable or
```
Comment (taion) [inline src/components/message.js:69]: lol oh gosh

File: src/components/withIntl.js [hunk 5]
```diff
@@ -1,69 +1,32 @@
-import React, {Component} from 'react';
-import hoistNonReactStatics from 'hoist-non-react-statics';
-import invariant from 'invariant';
-import createContext from 'create-react-context';
-import {invariantIntlContext} from '../utils';
+import React from 'react';
+import hostNonReactStatics from 'hoist-non-react-statics'
```
Comment (taion) [inline src/components/withIntl.js]: `hoistNonReactStatics`

**Review by taion — APPROVED** [review]
i feel like removing the semicolons here made this like 40% harder to review

File: src/components/message.js (hunk 3 above)
Comment (jquense) [reply src/components/message.js:78]: no, but i was trying to do a minimal port as first pass

File: src/components/date.js (hunk 2 above)
Comment (jquense) [reply src/components/date.js:28]: That was my thought for the future, this is only one with a lot tho so it seemed fine to do long hand

**Comment by jquense** [comment]
Bah i didn't realize that happened, the prettier config seems to only be in the script itself, dumb

**Pull request merged**